File: PageObjectDotNet/AutomationPractice/PageObjects/AutomationPracticeStore/Objects/MiniCartItem.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;
using OpenQA.Selenium.Support.UI;
using SeleniumFramework.Base;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace AutomationPractice.PageObjects.AutomationPracticeStore.Objects
{
    public class MiniCartItem
    {
        [FindsBy(How = How.CssSelector, Using = "a.cart-images img")]
        private IWebElement thumbnailPicture;

        [FindsBy(How = How.ClassName, Using = "quantity")]
        private IWebElement quantity;

        [FindsBy(How = How.ClassName, Using = "cart_block_product_name")]
        private IWebElement productName;

        [FindsBy(How = How.ClassName, Using = "product-atributes")]
        private IWebElement attributes;

        [FindsBy(How = How.ClassName, Using = "price")]
        private IWebElement total;

        [FindsBy(How = How.ClassName, Using = "ajax_cart_block_remove_link")]
        private IWebElement buttonClose;


        private readonly IWebElement item;

        public MiniCartItem(IWebElement miniItem) 
        {
            item = miniItem;
        }

        public string GetThumbnailImageUrl() 
        {
            return thumbnailPicture.GetAttribute("src");
        }

        public int GetQuantity() 
        {
            return int.Parse(quantity.Text);
        }

        public string GetProductFullName() 
        {
            return productName.GetAttribute("title");
        }

        public ReadOnlyCollection<string> GetProductAttributes() 
        {
            var attrs = attributes.Text.Replace(" ", string.Empty).Split(',');
            return attrs.ToList().AsReadOnly();
        }

        public double GetProductTotalPrice() 
        {
            return double.Parse(total.Text);
        }

        public void RemoveItem()
        {
            var wait = new WebDriverWait(TestConfiguration.GetDriverInstance(), TimeSpan.FromSeconds(3));
            buttonClose.Click();
            wait.Until(ExpectedConditions.StalenessOf(item));
        }

    }
}

File: PageObjectJava/src/test/java/PageObjects/AutomationPracticeStore/Objects/MiniCartItem.java
package PageObjects.AutomationPracticeStore.Objects;

import Base.TestConfiguration;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MiniCartItem {

    @FindBy(how = How.CSS, using = "a.cart-images img")
    private WebElement thumbnailPicture;

    @FindBy(className = "quantity")
    private WebElement quantity;

    @FindBy(how = How.CLASS_NAME, using = "cart_block_product_name")
    private WebElement productName;

    @FindBy(how = How.CLASS_NAME, using = "product-atributes")
    private WebElement attributes;

    @FindBy(className = "price")
    private WebElement total;

    @FindBy(className = "ajax_cart_block_remove_link")
    private WebElement closeButton;

    private final WebElement miniItem;

    public MiniCartItem(WebElement miniItem){
        this.miniItem = miniItem;
    }

    public String getThumbnailImageLink(){
        return this.thumbnailPicture.getAttribute("src");
    }

    public int getQuantity(){
        return Integer.parseUnsignedInt(this.quantity.getText());
    }

    public String getProductFullName(){
        return this.productName.getAttribute("title");
    }

    public List<String> getProductAttributes(){
        List<String> attributeList = new ArrayList<>();
        String txt = this.attributes.getText();
        Collections.addAll(attributeList, txt.split(", "));
        return attributeList;
    }

    public double getProductTotalPrice(){
        return Double.parseDouble(this.total.getText());
    }

    public void removeItem(){
        WebDriverWait wait = new WebDriverWait(TestConfiguration.getDriver(), 3);
        closeButton.click();
        wait.until(ExpectedConditions.stalenessOf(this.miniItem));
    }


}
